### PR TITLE
Added moore's voting algorithm

### LINKED ISSUE #245: [Feature Request] moore's voting algorithm
i want to add moore's voting algorithm.
it basically finds the majority element of the array
@ajay-dhangar please assign me with level 2, gssoct-extd, hacktoberfest

### PR DESCRIPTION
**Pull request by meghanakn473**
📥 Pull Request

**Description**
This PR adds documentation for Moore's Voting Algorithm to the docs/moores-voting-algorithm directory. It includes an overview, characteristics, complexity analysis, and a C++ implementation to help users understand and implement this efficient algorithm for finding the majority element in an array.

Fixes https://github.com/ajay-dhangar/algo/issues/245

**Type of change**

[  ] Bug fix (non-breaking change which fixes an issue) 
[X ] New feature (non-breaking change which adds functionality)
[  ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
[X ] Documentation update

**Checklist**

[ X] My code follows the style guidelines of this project
[ X] I have performed a self-review of my code
[ X] I have commented my code, particularly in hard-to-understand areas
[ X] I have added tests that prove my fix is effective or that my feature works
[ X] New and existing unit tests pass locally with my changes
[ X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by meghanakn473** [comment]
@ajay-dhangar i have re made pull request by synck fork and updating the last commits done. 

please review and merge

**Comment by meghanakn473** [comment]
@ajay-dhangar please assign me level 2 for this pr (as i added 2 algo in single pr, while assigning the issuse for both algo u gave separetly level 1 for each) , also please review and merge asap.

i hope ull consider it.

File: docs/Kadane-Algo/kadane-algo.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+id: kadanes-algorithm
```
Comment (ajay-dhangar) [inline docs/Kadane-Algo/kadane-algo.md:2]: add on top ``` --- ```

File: docs/MooreVotingAlgo/moore-voting-algo.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
+id: moores-voting-algorithm
```
Comment (ajay-dhangar) [inline docs/MooreVotingAlgo/moore-voting-algo.md:2]: add on top ``` --- ``

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: docs/Kadane-Algo/kadane-algo.md (hunk 1 above)
Comment (meghanakn473) [reply docs/Kadane-Algo/kadane-algo.md:2]: > add on top `---`

Done!!

**Comment by meghanakn473** [comment]
> update your code based on suggestions

i made changes kindly review and merge @ajay-dhangar

**Comment by meghanakn473** [comment]
@ajay-dhangar please allow merge

**Comment by meghanakn473** [comment]
@ajay-dhangar please check

**Review by ajay-dhangar — APPROVED** [review]
All is Done

**Pull request merged**